### PR TITLE
feature/262: Added automatic fix for the version of parent pom 

### LINKED ISSUE #262: Automatically fix version of partent pom ref
Automatically fix version of `pk_generated_partent.pom` reference in `pom.xml`.

### PR DESCRIPTION
**Pull request by jakobbraun**
262: Added automatic fix for the version of parent pom reference in the pom.xml
Closes #262

### CODE REVIEW

File: doc/changes/changes_2.3.0.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Project Keeper Project 2.3.0, released 2022-??-??
```
Comment (Nicoretti) [inline doc/changes/changes_2.3.0.md:1]: missing release date?

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileValidator.java [hunk 2]
```diff
@@ -14,14 +14,14 @@
 import com.exasol.projectkeeper.Validator;
 import com.exasol.projectkeeper.config.ProjectKeeperConfig;
 import com.exasol.projectkeeper.validators.files.RequiredFileValidator;
-import com.exasol.projectkeeper.validators.finding.SimpleValidationFinding;
-import com.exasol.projectkeeper.validators.finding.ValidationFinding;
+import com.exasol.projectkeeper.validators.finding.*;
```
Comment (Nicoretti) [inline project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileValidator.java:17]: not sure about `Java` but in `Python` I generally would discourage `*` imports

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileValidator.java [hunk 3]
```diff
@@ -57,12 +57,28 @@ public PomFileValidator(final Path projectDirectory, final Collection<ProjectKee
...
+        if (hasFindingWithFix(findings)) {
+            return List.of(
+                    new ValidationFindingGroup(findings, () -> new PomFileIO().writePomFile(pom, this.pomFilePath)));
+        } else {
+            return findings;
+        }
+    }
+
+    private boolean hasFindingWithFix(final List<ValidationFinding> findings) {
+        return findings.stream().anyMatch(finding -> //
```
Comment (Nicoretti) [inline project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileValidator.java:77]: `//` comment left over?

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileValidator.java [hunk 4]
```diff
@@ -57,12 +57,28 @@ public PomFileValidator(final Path projectDirectory, final Collection<ProjectKee
...
+        (finding instanceof ValidationFindingGroup) || //
```
Comment (Nicoretti) [inline project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileValidator.java:78]: `//` comment left over?

File: project-keeper/src/test/java/com/exasol/projectkeeper/validators/pom/PomFileValidatorTest.java [hunk 5]
```diff
@@ -102,6 +109,21 @@ void testMissingGroupId() throws IOException {
...
+    void testFixParentVersion() throws IOException, XmlPullParserException {
+        final Parent parentWithWrongVersion = new Parent();
+        parentWithWrongVersion.setArtifactId("my-test-project-generated-parent");
+        parentWithWrongVersion.setGroupId(TestMavenModel.PROJECT_GROUP_ID);
+        parentWithWrongVersion.setVersion("3.2.0");
+        parentWithWrongVersion.setRelativePath("./pk_generated_parent.pom");
+        final TestMavenModel model = new TestMavenModel();
+        model.setParent(parentWithWrongVersion);
+        model.writeAsPomToProject(this.tempDir);
+        runFix(null);
```
Comment (Nicoretti) [inline project-keeper/src/test/java/com/exasol/projectkeeper/validators/pom/PomFileValidatorTest.java:122]: Is  `runFix(null)` a method of the fixture? Looks odd seems to have a side effect on the test but neither gets input nor output refs.

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileValidator.java (hunk 2 above)
Comment (jakobbraun) [reply project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileValidator.java:17]: That's automated by our formatter. After a certain amount of imports it starts grouping.  @redcatbear created a config according to our style.

File: project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileValidator.java (hunk 3 above)
Comment (jakobbraun) [reply project-keeper/src/main/java/com/exasol/projectkeeper/validators/pom/PomFileValidator.java:77]: We do that for manual formatting. By adding `//` the formatter can't remove the linebreak...

File: project-keeper/src/test/java/com/exasol/projectkeeper/validators/pom/PomFileValidatorTest.java (hunk 5 above)
Comment (jakobbraun) [reply project-keeper/src/test/java/com/exasol/projectkeeper/validators/pom/PomFileValidatorTest.java:122]: yes, it runs on the tempDir. The tempdir is a temporary directory and the ref is stored as local field. So runFix can access it wiothout a ref.

File: doc/changes/changes_2.3.0.md (hunk 1 above)
Comment (jakobbraun) [reply doc/changes/changes_2.3.0.md]:
```suggestion
# Project Keeper Project 2.3.0, released 2022-03-28
```

**Review by Nicoretti — APPROVED** [review]

**Pull request merged**